desktop/ui/src/main/java/org/datacleaner/windows/ElasticSearchDatastoreDialog.java

```diff
@@ -47,25 +58,129 @@
 
     private static final long serialVersionUID = 1L;
 
+    private static final ElasticSearchDatastore.ClientType DEFAULT_CLIENT_TYPE = ElasticSearchDatastore.ClientType.TRANSPORT;
+    private static final boolean DEFAULT_SSL = false;
+
+    private final JComboBox<ClientType> _clientTypeComboBox;
     private final JXTextField _hostnameTextField;
     private final JXTextField _portTextField;
     private final JXTextField _clusterNameTextField;
     private final JXTextField _indexNameTextField;
+    private final JXTextField _usernameTextField;
+    private final JPasswordField _passwordField;
+    private final JCheckBox _sslCheckBox;
+    private final FilenameTextField _keystorePathField;
+    private final JPasswordField _keystorePasswordField;
 
     @Inject
     public ElasticSearchDatastoreDialog(WindowContext windowContext, MutableDatastoreCatalog catalog,
-            @Nullable ElasticSearchDatastore originalDatastore, UserPreferences userPreferences) {
+            @Nullable final ElasticSearchDatastore originalDatastore, UserPreferences userPreferences) {
         super(originalDatastore, catalog, windowContext, userPreferences);
 
         setSaveButtonEnabled(false);
-        
+
+        _clientTypeComboBox = new JComboBox<>(ClientType.values());
+        _clientTypeComboBox.setSelectedItem(DEFAULT_CLIENT_TYPE);
+
+        // Both NODE and TRANSPORT
+        _clusterNameTextField = WidgetFactory.createTextField();
+        _indexNameTextField = WidgetFactory.createTextField();
+
+        // Only TRANSPORT
         _hostnameTextField = WidgetFactory.createTextField();
         _portTextField = WidgetFactory.createTextField();
         _portTextField.setDocument(new NumberDocument(false));
-        _clusterNameTextField = WidgetFactory.createTextField();
-        _indexNameTextField = WidgetFactory.createTextField();
+        _usernameTextField = WidgetFactory.createTextField();
+        _passwordField = WidgetFactory.createPasswordField();
+        _keystorePathField = new FilenameTextField(userPreferences.getOpenDatastoreDirectory(), true);
+        // FIXME: Hack-ish way to make it fit...
+        final double columns = WidgetFactory.TEXT_FIELD_COLUMNS * 0.6;
```

Comment:
OK. Strange. This is not a big concern, but maybe something to spend an extra 10 minutes on checking (if it works for Excel it ought to be the same here - why isn't it?)